cmd/kubectl-gadget/biolatency.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2019-2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/kinvolk/inspektor-gadget/cmd/kubectl-gadget/utils"
+	gadgetv1alpha1 "github.com/kinvolk/inspektor-gadget/pkg/api/v1alpha1"
+)
+
+const (
+	GADGET_NAME        = "biolatency"
+	GADGET_OUTPUTMODE  = "Status"
+	GADGET_OUTPUTSTATE = "Completed"
+)
+
+var biolatencyTraceConfig = &utils.TraceConfig{
+	GadgetName:                GADGET_NAME,
+	TraceOutputMode:           GADGET_OUTPUTMODE,
+	TraceOutputState:          GADGET_OUTPUTSTATE,
+	CommonFlags:               &params,
+	IgnoreKubeConfigNamespace: true,
+}
+
+var biolatencyLogger *log.Entry
+
+var biolatencyCmd = &cobra.Command{
+	Use:   "biolatency",
+	Short: "Generate a histogram with the distribution of block device I/O latency",
+}
+
+var biolatencyStartCmd = &cobra.Command{
+	Use:    "start",
+	Short:  "Start monitor the block device I/O (disk I/O) and records the distribution of I/O latency (time)",
+	Run:    runBiolatencyStart,
+	PreRun: setContextLogger,
+}
+
+var biolatencyStopCmd = &cobra.Command{
+	Use:    "stop",
+	Short:  "Stop monitoring and generate a report (a histogram graph) with the distribution of block device I/O latency",
+	Run:    runBiolatencyStop,
+	PreRun: setContextLogger,
+}
+
+var biolatencyListCmd = &cobra.Command{
+	Use:    "list",
+	Short:  "List the currently running biolatency traces",
+	Run:    runBiolatencyList,
+	PreRun: setContextLogger,
+}
+
+func init() {
+	biolatencyCmd.AddCommand(biolatencyStartCmd)
+	biolatencyCmd.AddCommand(biolatencyStopCmd)
+	biolatencyCmd.AddCommand(biolatencyListCmd)
+
+	utils.AddCommonFlags(biolatencyCmd, &params)
+	rootCmd.AddCommand(biolatencyCmd)
+}
+
+func runBiolatencyStart(cmd *cobra.Command, args []string) {
+	if params.Node == "" {
+		biolatencyLogger.Fatalf("%s only works with --node\n", GADGET_NAME)
+	}
+
+	biolatencyTraceConfig.Operation = "start"
+	globalTraceID, err := utils.CreateTrace(biolatencyTraceConfig)
+	if err != nil {
+		biolatencyLogger.Fatalf("Error creating trace: %v\n", err)
+	}
+
+	biolatencyLogger.Infof("Use the following ID to stop the gadget and get the report: %s\n", globalTraceID)
```

Comment:
At first, I added a long information message like this, then I realized sometime it is not needed, so I modified my code like this:
```golang
// quiet is BoolVarP given on command line with --quiet.
if !quiet {
		fmt.Printf("Succesfully created trace with ID: %q!\nYou will need to use this ID to report trace.\nIn case you forgot it, you can list existing traces with 'list' subcommand\n", traceID)
	} else {
		fmt.Printf("%s\n", traceID)
	}
```
https://github.com/kinvolk/inspektor-gadget/pull/380/commits/f416a17ef53e0931ceee10e2c72e67338ff83960#diff-a886de67c1b9577e3f404f9fe50aaea77c177bd7b2cc682bf1ce3e9028188be5R91

Reply:
Yeah, I was also wondering what is the best way to show that info to the user. Your proposal could be a solution.

cc @mauriciovasquezbernal any opinion regarding this?

Reply:
I trying to make my head around this ID mechanism. I think printing that whole message to the user is very annoying, perhaps it's useful the first time but then it's just a lot of trash there. 

I'm trying to look at other cases that use the same mechanism, for instance Docker just prints the Container ID without any other info:

```
$ docker run -d ubuntu sleep 10000 
3c59124b3070acbe136c704e6134f9969f9aa1fe1976abbbf4135159ea5ce1c5
```

Regarding this specific point, I'll just print the ID and remove the `quiet` flag.

Reply:
I agree with you, thus I removed it from my PR.